### PR TITLE
Moved the `mesos-modules` to build of 1.8.x branch.

### PR DESCRIPTION
**Pull request by asridharan**
High level description including:
 - What does this change enable
We have a version specific branch for `mesos-modules`. We want DC/OS mesos-modules packages to track the corresponding version in the `dcos-mesos-mdoules` branch.
 - What bugs does this change fix
[DCOS-619](https://dcosjira.atlassian.net/browse/DCOS-619)
 - High-Level how things changed
Mesos modules compiled against a version specific branch.
# Issues

[DCOS-619](https://dcosjira.atlassian.net/browse/DCOS-619)
[DCOS-13192](https://mesosphere.atlassian.net/browse/DCOS-13192)
[DCOS-13194](https://mesosphere.atlassian.net/browse/DCOS-13194)

# Checklist

 - [X] Included a test which will fail if code is reverted but test is not
 - [X] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

If this is a component/package update, include

 - [ ] Change Log from last: <link>
 - [X] Test Results:
   https://jenkins.mesosphere.com/service/jenkins/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/29/console
There are a few integration tests that are also part of this PR and should be tested using the upgrade integration tests.
 - [X] Code Coverage: N/A

### CODE REVIEW

**Comment by adammesos** [comment]
Adding the WIP label for now. CI is green, but let us know when it's "Ready for Review" and @mention or assign reviewers.

**Comment by asridharan** [comment]
@adam-mesos we should be ready to review this. The cherry-picks for the dcos_config.yaml and integration tests are also part of this PR. 

@orsenthil how do we run the upgrade integration tests for this PR for 1.8.7 -> 1.8.8 . The integration tests that have been cherry-picked into this PR are supposed to get stressed during upgrades.

**Comment by asridharan** [comment]
@adam-mesos @kaysoky we can review this. 

The main pending task is to run upgrade tests on this PR <- @orsenthil

**Comment by adammesos** [comment]
Could you please update the "component checklist" with the changelog/diff for the dcos-mesos-modules bump and the Jenkins CI test results?

**Comment by asridharan** [comment]
@adammesos we don't have Jenkins CI for the mesos-modules (we didn't port that job at this point to the new repo). Though added some unit-tests cases in the overlay modules and integration tests to test the endpoints. The integrations tests are getting exercised here. Will add data on the unit-test.

**Comment by adammesos** [comment]
We do have a nightly build for dcos-mesos-modules:
https://jenkins.mesosphere.com/service/jenkins/view/Mesos/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/
I'm sure @karya0 can help you do a custom build of these against a specific dcos-mesos SHA for 1.8.8
I'd love to see a test run of these before we merge, so I don't have to trust that you ran the unit tests.

**Comment by asridharan** [comment]
Looks like the upgrade tests are failing. Had a chat with @orsenthil and seems like there is some infrastructure issue because of which the upgrade tests are not able to run.

**Comment by asridharan** [comment]
@karya0 can we run the CI job https://jenkins.mesosphere.com/service/jenkins/view/Mesos/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/ for the 1.8.x branch on the `dcos-mesos-modules` repo?

Thanks

**Comment by orsenthil** [comment]
Yeah, the upgrade test failures are due the Infrastructure problem. We will setup the agents which will be appropriate for this testing this 1.8.8 upgrade.

**Comment by asridharan** [comment]
@adam-mesos updated the test results section. @karya0 was kind enough to setup a parameterized build for the `dcos-mesos-modules` repo.

**Comment by adammesos** [comment]
Looks like https://jenkins.mesosphere.com/service/jenkins/view/Mesos/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/17/ is actually the 1.8.x custom build (since the 'latest' nightly is from master). That build fails two tests that shouldn't exist in 1.8.x (journald and logsink).

File: packages/mesos/build [hunk 1]
```diff
@@ -20,7 +20,14 @@ LIBS="-lssl -lcrypto" LDFLAGS="-L/opt/mesosphere/active/openssl/lib" "/pkg/src/m
 make -j$NUM_CORES
 
 make install
+
 cp src/java/target/mesos-*.jar "$libdir"
+
+# This is copied during `make install` as per the configure flag
+# `--enable-install-module-dependencies`.  Modules that have their
+# own protobufs may wish to use this during their builds.
+ln -s "$PKG_PATH/lib/mesos/3rdparty/bin/protoc" "$PKG_PATH/bin/protoc"
```
Comment (adammesos) [inline packages/mesos/build]: Do any of our modules actually use this in 1.8.x?

File: packages/mesos-modules/buildinfo.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "single_source" : {
       "kind": "git",
       "git": "https://github.com/dcos/dcos-mesos-modules.git",
-      "ref": "8d09c2d44f606df23a57254ebcbc5f2722a3b677",
-      "ref_origin": "master"
+      "ref": "86f14f9402e324f78385a0b7b9509079dda7c0ee",
+      "ref_origin": "1.8.x"
```
Comment (adammesos) [inline packages/mesos-modules/buildinfo.json]: Looking at these SHAs, the modules changelog is actually: https://github.com/dcos/dcos-mesos-modules/compare/8d09c2d44f606df23a57254ebcbc5f2722a3b677...86f14f9402e324f78385a0b7b9509079dda7c0ee

File: packages/mesos-modules/build [hunk 3]
```diff
@@ -5,11 +5,19 @@ popd
...
+# Add the 3rdparty directories.
+export CPPFLAGS='-I/opt/mesosphere/lib/mesos/3rdparty/include'
+export CFLAGS=-I/opt/mesosphere/lib/mesos/3rdparty/include
+export LDFLAGS='-L/opt/mesosphere/lib/mesos/3rdparty/lib'
+
 mkdir -p build
 pushd build
 /pkg/src/mesos-modules/configure \
      --with-mesos=/opt/mesosphere/active/mesos \
+     --with-openssl=/opt/mesosphere/active/openssl \
```
Comment (adammesos) [inline packages/mesos-modules/build]: Do these modules use openssl for anything?

File: packages/mesos-modules/buildinfo.json (hunk 2 above)
Comment (adammesos) [inline packages/mesos-modules/buildinfo.json]: Why does the new dcos-mesos-modules self-identify as `AC_INIT([dcos-mesos-modules],[0.2])` (same as what's in master/1.9)? Should this be 0.1 instead, or 1.8.8? How do we want to version these modules? I'd suggest we follow along with the DC/OS version number they're targeted for (1.8.8 vs. 1.9.0)

Comment (adammesos) [inline packages/mesos-modules/buildinfo.json]: There are also a few parts of the README and configure.ac that reference the systemd journald headers. Those are unnecessary for 1.8.8

**Comment by asridharan** [comment]
@adam-mesos my bad. Looks like I copy pasted the master's console. Have updated the link to the right console output for build #18. Note in that build it will still report two failures for log sink module, that's because the log sink module doesn't exist in that branch and @karya0 felt since we are going to do these one off builds for these branches didn't make sense to update the Jenkins scripts.

File: packages/mesos/build (hunk 1 above)
Comment (asridharan) [inline packages/mesos/build]: yes we need protobuf generation for the overlay modules. The compilation will fail if we don't have this.

File: packages/mesos-modules/buildinfo.json (hunk 2 above)
Comment (asridharan) [inline packages/mesos-modules/buildinfo.json]: I agree. Thanks, will update this 1.8.8 and 1.9.0 for the Master.

File: packages/mesos-modules/build (hunk 3 above)
Comment (asridharan) [inline packages/mesos-modules/build]: Probably my lack of knowledge. These were included in 1.9 so added it here. @kaysoky can you shed some light as to which module is using openssl? In DC/OS the overlay modules don't necessarily need it, apart from when the master agent comm needs to happen over https but that I am assuming is a libprocess construct and `libmesos` should already be compiled with `openssl`.

File: packages/mesos-modules/buildinfo.json (hunk 2 above)
Comment (asridharan) [inline packages/mesos-modules/buildinfo.json]: @adam-mesos I see the systemd references in README.md, but I thought removed all references to systemd and journald headers from `configure.ac`? Could you point to any that you might have found in `configure.ac`?

Comment (adammesos) [inline packages/mesos-modules/buildinfo.json]: I can't find the systemd header reference in configure.ac anymore. I'm not sure what I was looking at, but it's gone now.

**Comment by adammesos** [comment]
Please let me know when the SHA has been bumped for the dcos-mesos-modules version string update, and I'll review again, and merge it, if there aren't any more concerns/pending-changes.

**Comment by asridharan** [comment]
@adammesos have comitted the changes to `dcos-mesos-modules`:
https://github.com/dcos/dcos-mesos-modules/commit/171504d8026770913db7c65cad708e8d9a52e762

Have pushed the SHA to DC/OS since the Jenkins build isn't passing yet. The AMI for this test needs `ipset`. @karya0 had put an `apt-get install ipset` in the build script but that doesn't seem to be there any more. Waiting for @karya0 to resolve the Jenkins issue and then will update the tests and push a new SHA to this PR.

**Comment by asridharan** [comment]
@adammesos I think we should be good to go here. Updated the test results with the latest Jenkins build. @karya0 has updated the AMI so that we have all the required dependencies.

**Comment by asridharan** [comment]
Spoke too soon :(. We need a patch for https://mesosphere.atlassian.net/browse/DCOS-11704 as well. Have added the fix to `dcos-mesos-modules/1.8.x` . Have a Jenkins build running on it
https://jenkins.mesosphere.com/service/jenkins/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/30/

Will update once I have this passed.

**Pull request merged**